pkg/providers/tinkerbell/template.go

```diff
@@ -69,9 +69,16 @@ func (tb *TemplateBuilder) GenerateCAPISpecControlPlane(clusterSpec *cluster.Spe
 	if err != nil {
 		return nil, err
 	}
+	var OSImageURL string
+	if clusterSpec.TinkerbellDatacenter.Spec.OSImageURL != "" {
+		OSImageURL = clusterSpec.TinkerbellDatacenter.Spec.OSImageURL
+	} else {
+		OSImageURL = tb.controlPlaneMachineSpec.OSImageURL
+	}
```

Comment:
I think we should preference the `OSImageURL` specified on the machine config, not the datacenter config.

Reply:
there's no precedence here actually as we don't allow specifying at both levels. so here we just set the value for generating template based on which level the value is set and it would never be the case that if machine config imageURL is present DC could take preference (as we error out early).!